api/base/serializers.py

```diff
@@ -249,7 +249,7 @@ def to_internal_value(self, data):
             type_ = self.root.Meta.type_
 
         if type_ != data:
-            raise Conflict()
+            raise Conflict(detail=('The resource type you specified "' + data + '" does not match the type of the resource you specified "' + type_ + '".'))
```

Comment:
This could probably still stand to be more useful. `This resource has a type of "collections", but you set the json body's type field to "collection". You probably need to change the type field to match the resource's type.`